staging/src/k8s.io/kube-aggregator/pkg/controllers/status/remote/remote_available_controller.go

```diff
@@ -239,30 +240,31 @@ func (c *AvailableConditionController) sync(key string) error {
 			return err
 		}
 
-		endpoints, err := c.endpointsLister.Endpoints(apiService.Spec.Service.Namespace).Get(apiService.Spec.Service.Name)
-		if apierrors.IsNotFound(err) {
-			availableCondition.Status = apiregistrationv1.ConditionFalse
-			availableCondition.Reason = "EndpointsNotFound"
-			availableCondition.Message = fmt.Sprintf("cannot find endpoints for service/%s in %q", apiService.Spec.Service.Name, apiService.Spec.Service.Namespace)
-			apiregistrationv1apihelper.SetAPIServiceCondition(apiService, availableCondition)
-			_, err := c.updateAPIServiceStatus(originalAPIService, apiService)
-			return err
-		} else if err != nil {
+		endpointSliceSelector := labels.SelectorFromSet(labels.Set{discoveryv1.LabelServiceName: apiService.Spec.Service.Name})
+		endpointSlices, err := c.endpointSliceLister.EndpointSlices(apiService.Spec.Service.Namespace).List(endpointSliceSelector)
+		if err != nil {
 			availableCondition.Status = apiregistrationv1.ConditionUnknown
 			availableCondition.Reason = "EndpointsAccessError"
 			availableCondition.Message = fmt.Sprintf("service/%s in %q cannot be checked due to: %v", apiService.Spec.Service.Name, apiService.Spec.Service.Namespace, err)
 			apiregistrationv1apihelper.SetAPIServiceCondition(apiService, availableCondition)
 			_, err := c.updateAPIServiceStatus(originalAPIService, apiService)
 			return err
+		} else if len(endpointSlices) == 0 {
+			availableCondition.Status = apiregistrationv1.ConditionFalse
+			availableCondition.Reason = "EndpointsNotFound"
+			availableCondition.Message = fmt.Sprintf("cannot find endpointslices for service/%s in %q", apiService.Spec.Service.Name, apiService.Spec.Service.Namespace)
+			apiregistrationv1apihelper.SetAPIServiceCondition(apiService, availableCondition)
+			_, err := c.updateAPIServiceStatus(originalAPIService, apiService)
+			return err
 		}
 		hasActiveEndpoints := false
 	outer:
-		for _, subset := range endpoints.Subsets {
-			if len(subset.Addresses) == 0 {
+		for _, slice := range endpointSlices {
+			if len(slice.Endpoints) == 0 {
 				continue
 			}
-			for _, endpointPort := range subset.Ports {
-				if endpointPort.Name == portName {
+			for _, endpointPort := range slice.Ports {
+				if endpointPort.Name != nil && *endpointPort.Name == portName {
 					hasActiveEndpoints = true
```

Comment:
do we also need to ensure at least one of `slice.Endpoints[i].Conditions.Ready` is not explicitly `false`?